sound/soc/bcm/hifiberry_dacplus.c

```diff
@@ -396,6 +454,30 @@ static int snd_rpi_hifiberry_dacplus_probe(struct platform_device *pdev)
 						"hifiberry-dacplus,slave");
 		leds_off = of_property_read_bool(pdev->dev.of_node,
 						"hifiberry-dacplus,leds_off");
+		auto_mute = of_property_read_bool(pdev->dev.of_node,
+						"hifiberry-dacplus,auto_mute");
+
+		/* check for HW MUTE as defined in DT-overlay
+		   active high, therefore default to HIGH to MUTE  */
+		snd_mute_gpio =	devm_gpiod_get_optional(&pdev->dev, "mute", GPIOD_OUT_HIGH);
+		if (snd_mute_gpio)
+			dev_info(&pdev->dev, "GPIO%i for HW-MUTE selected",
```

Comment:
An error is still a possible outcome that should be reported here - an absent property results in a NULL, but some other problem will still cause an error (which is probably what you want).